### PR TITLE
NJ 47 - add w2 wages sum to PDF and XML

### PR DESCRIPTION
**Pull request by aloverso**
## Link to pivotal/JIRA issue
- https://github.com/newjersey/affordability-pm/issues/47

## Is PM acceptance required? (delete one)
- Yes - don't merge until issue is accepted!

## What was done?
- Sum `StateWagesAmt` from all W2s from federal DF, and round to nearest whole number
- Output sum in NJ XML field `WagesSalariesTips`
- Output sum in Line 15 of PDF
- [code] update zeus_two_w2s and zeus_many_w2s to use more interesting wage numbers with decimals

## How to test?
- Use `zeus_minimal` for a user with no DF W2s to confirm XML are not filled in when no W2
- Use `zeus_two_w2s` to test with wages: `12345.67` + `50000`
- Use `zeus_many_w2s` to test with wages: `50000.33` + `50000.33` + `50000.33` + `50000.33`
  
## Screenshots (for visual changes)
image
image

### CODE REVIEW

File: app/models/state_file_nj_intake.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@
 #  last_sign_in_ip                   :inet
 #  locale                            :string           default("en")
 #  locked_at                         :datetime
+#  love_taxes                        :string
```
[the remark this answers is not in the document]
Comment (aloverso) [reply app/models/state_file_nj_intake.rb:32]: 🤦

File: app/lib/efile/nj/nj1040.rb [hunk 2]
```diff
@@ -55,6 +56,19 @@ def analytics_attrs
...
       private
 
+      def calculate_line_15
+        if @direct_file_data.w2s.empty?
+          return -1
+        end
+
+        sum = 0
+        @direct_file_data.w2s.each do |w2|
+          state_wage = w2.node.at("W2StateLocalTaxGrp StateWagesAmt").text.to_f
```
Comment (jenny-heath) [inline app/lib/efile/nj/nj1040.rb:66]: [PEBBLE] you can use the accessor on `DfW2` here https://github.com/codeforamerica/vita-min/blob/main/app/models/df_w2_accessor.rb#L18

so it would look like `state_wage = w2.StateWagesAmt.to_f`

Comment (aloverso) [reply app/lib/efile/nj/nj1040.rb:66]: Replied in slack, agreed we're not changing for now

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb [hunk 3]
```diff
@@ -402,5 +402,79 @@
         end
       end
     end
+
+    describe "line 15 wages" do
```
Comment (jachan) [inline spec/lib/pdf_filler/nj1040_pdf_spec.rb:406]: [SAND] Is it reasonable to try a test case where we have an overflow? e.g. W2 wages sum to 10000000000.00? I understand that someone earning that much would probably be screened out of our tool but I'm curious how our application would handle a case like that.

**Review by jachan — APPROVED** [review]
Looks good! Left a comment about an edge case (very big W2 sum).

**Comment by embarnard** [comment]
This code all looked good to me, excellent work! Obviously there was a test that was failed when merged to main but doubtful we would have caught that since it didn't show in circle. 

No biggie but for future practice, I think its important to have an explicit acceptance from our side (meaning you get an actual approved status from github) before merging the PR even if it seems that someone else might have looked at the PR or left a comment. Sometimes we might start looking at a PR but get pulled away to other things but will usually try to get back to it by EOD. @aloverso @jachan

File: spec/lib/pdf_filler/nj1040_pdf_spec.rb (hunk 3 above)
Comment (aloverso) [reply spec/lib/pdf_filler/nj1040_pdf_spec.rb:406]: Nope, no need! You can see a conversation about this question in the story, and also more context [in this slack thread](https://njcio.slack.com/archives/C07GYAQ4E1E/p1726515777378459)
